dask/dataframe/shuffle.py

```diff
@@ -25,21 +28,40 @@
 
 
 def _calculate_divisions(
-    df,
-    partition_col,
-    repartition,
-    npartitions,
-    upsample=1.0,
-    partition_size=128e6,
-):
+    df: DataFrame,
+    partition_col: Series,
+    repartition: bool,
+    npartitions: int,
+    upsample: float = 1.0,
+    partition_size: float = 128e6,
+) -> Tuple[List, List, List]:
     """
     Utility function to calculate divisions for calls to `map_partitions`
     """
     sizes = df.map_partitions(sizeof) if repartition else []
     divisions = partition_col._repartition_quantiles(npartitions, upsample=upsample)
     mins = partition_col.map_partitions(M.min)
     maxes = partition_col.map_partitions(M.max)
-    divisions, sizes, mins, maxes = compute(divisions, sizes, mins, maxes)
+    nulls_presence = partition_col.map_partitions(M.isna)
+
+    try:
+        divisions, sizes, mins, maxes, nulls_presence = compute(
+            divisions, sizes, mins, maxes, nulls_presence
+        )
+    except Exception as e:
+        # Check if there are nulls and if so, inform the user about this probably being the cause behing the error
+        if nulls_presence.any().compute() and not is_numeric_dtype(partition_col.dtype):
```

Comment:
Thanks for making this change @jorloplaz! I am wondering if we can tell from the error message if there were nans rather than having to do another compute before raising.

Reply:
Not from the error message, because different things can happen. For example, we could have an error like:

> File "/home/jorge.lopez/anaconda3/envs/AML-dev/lib/python3.9/site-packages/numpy/core/_methods.py", line 39, in _amax
>     return umr_maximum(a, axis, None, out, keepdims, initial, where)
> TypeError: '>=' not supported between instances of 'float' and 'str'

But also I've found things like:

> dask/array/percentile.py:40: in _percentile
>     result[0] = min(result[0], values.min())
> TypeError: Cannot convert NaTType to pandas._libs.tslibs.timestamps._Timestamp

So what I did is to change the general `Exception` for a `TypeError, without checking the message itself.